### PR TITLE
[prometheus-operator] Add missing features from stable

### LINKED ISSUE #2636: [prometheus-operator] Add Provision for Alertmanager Templates
**Which chart**:
bitnami/prometheus-operator chart version 0.17.1

**Is your feature request related to a problem? Please describe.**

The [stable/prometheus-operator](https://github.com/helm/charts/tree/master/stable/prometheus-operator) chart provides a way to supply Alertmanager templates and have them [included in the Alertmanager secret](https://github.com/helm/charts/blob/d6973a495463725d1a2ac67d56f0f834753e0eb6/stable/prometheus-operator/templates/alertmanager/secret.yaml#L20-L22) as decribed [here](https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/alerting.md#alerting). The bitnami chart does not, and without that, there is no convenient way to supply Alertmanager templates. 

**Describe the solution you'd like**

Provide feature parity with stable/prometheus-operator

<details><summary>Possible patch</summary>

```diff
diff --git a/bitnami/prometheus-operator/templates/alertmanager/secrets.yaml b/bitnami/prometheus-operator/templates/alertmanager/secrets.yaml
index 8ccdb1e3..fcd685aa 100644
--- a/bitnami/prometheus-operator/templates/alertmanager/secrets.yaml
+++ b/bitnami/prometheus-operator/templates/alertmanager/secrets.yaml
@@ -6,4 +6,7 @@ metadata:
   labels: {{- include "prometheus-operator.alertmanager.labels" . | nindent 4 }}
 data:
   alertmanager.yaml: {{ toYaml .Values.alertmanager.config | b64enc | quote }}
+{{- range $key, $val := .Values.alertmanager.templateFiles }}
+  {{ $key }}: {{ $val | b64enc | quote }}
+{{- end }}
 {{- end }}
```
</details>

**Describe alternatives you've considered**

I considered using `alertmanager.configMaps` but there is no example or description of how to use it (is it just a list of names of configMaps? where do they get mounted?) and I do not want to have to separately create a configMap.

### LINKED ISSUE #2637: [prometheus-operator] Allow setting externalUrl
**Which chart**:
bitnami/prometheus-operator chart version 0.17.1

**Is your feature request related to a problem? Please describe.**

Prometheus and Alertmanager [must be configured](https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/exposing-prometheus-and-alertmanager.md#kubernetes-api) with the full URL at which they will be exposed.
The [stable/prometheus-operator](https://github.com/helm/charts/tree/master/stable/prometheus-operator) chart provides a way to set `externalUrl` for [Prometheus](https://github.com/helm/charts/blob/d6973a495463725d1a2ac67d56f0f834753e0eb6/stable/prometheus-operator/templates/prometheus/prometheus.yaml#L52-L58) and [Alertmanager](https://github.com/helm/charts/blob/d6973a495463725d1a2ac67d56f0f834753e0eb6/stable/prometheus-operator/templates/alertmanager/alertmanager.yaml#L18-L24). The bitnami chart does not, and without that, there is no way to deploy Prometheus and Alertmanager behind a custom Ingress. 

**Describe the solution you'd like**

Provide feature parity with stable/prometheus-operator

<details><summary>Possible patch</summary>

```diff
diff --git a/bitnami/prometheus-operator/templates/alertmanager/alertmanager.yaml b/bitnami/prometheus-operator/templates/alertmanager/alertmanager.yaml
index 7cd13b19..13350eea 100644
--- a/bitnami/prometheus-operator/templates/alertmanager/alertmanager.yaml
+++ b/bitnami/prometheus-operator/templates/alertmanager/alertmanager.yaml
@@ -11,7 +11,9 @@ spec:
   image: {{ template "prometheus-operator.alertmanager.image" . }}
   {{- end }}
   listenLocal: {{ .Values.alertmanager.listenLocal }}
-  {{- if and .Values.alertmanager.ingress.enabled .Values.alertmanager.ingress.hosts }}
+  {{- if .Values.alertmanager.externalUrl }}
+  externalUrl: "{{ .Values.alertmanager.externalUrl }}"
+  {{- else if and .Values.alertmanager.ingress.enabled .Values.alertmanager.ingress.hosts }}
   externalUrl: {{ if .Values.alertmanager.ingress.tls }}https{{else}}http{{ end }}://{{ (index .Values.alertmanager.ingress.hosts 0).name }}/
   {{- else }}
   externalUrl: http://{{ template "prometheus-operator.alertmanager.fullname" . }}.{{ .Release.Namespace }}:{{ .Values.alertmanager.service.port }}
diff --git a/bitnami/prometheus-operator/templates/prometheus/prometheus.yaml b/bitnami/prometheus-operator/templates/prometheus/prometheus.yaml
index 1afc5885..5bd68ec4 100644
--- a/bitnami/prometheus-operator/templates/prometheus/prometheus.yaml
+++ b/bitnami/prometheus-operator/templates/prometheus/prometheus.yaml
@@ -40,7 +40,9 @@ spec:
   {{- else if .Values.prometheus.replicaExternalLabelName }}
   replicaExternalLabelName: "{{ .Values.prometheus.replicaExternalLabelName }}"
   {{- end }}
-  {{- if and .Values.prometheus.ingress.enabled .Values.prometheus.ingress.hosts }}
+  {{- if .Values.prometheus.externalUrl }}
+  externalUrl: "{{ .Values.prometheus.externalUrl }}"
+  {{- else if and .Values.prometheus.ingress.enabled .Values.prometheus.ingress.hosts }}
   externalUrl: {{ if .Values.prometheus.ingress.tls }}https{{else}}http{{ end }}://{{ (index .Values.prometheus.ingress.hosts 0).name }}/
   {{- else }}
   externalUrl: http://{{ template "prometheus-operator.prometheus.fullname" . }}.{{ .Release.Namespace }}:9090
```
</details>

### PR DESCRIPTION
**Pull request by Nuru**
**Description of the change**

- Allow setting externalUrl (fixes #2637)
- Allow adding Alertmanager Template Files to Alertmanager secret (fixes #2636)
- Allow setting a custom podAntiAffinity
- Allow setting sessionAffinity
- Allow annotations for ServiceAccounts

**Benefits**

- (Feature Parity with stable chart) Allow placement of Prometheus and/or Alertmanager behind a proxy
- (Feature Parity with stable chart) Allow provision of Alertmanger Templates directly
- (partial Feature Parity with stable chart) Allow other kinds of antiAffinity and topology keys
- (Feature Parity with stable chart) Allow session affinity for HA replicas to provide a stable user experience
- (Feature Parity with stable chart) Allow ServiceAccount to obtain IAM role in AWS

**Possible drawbacks**

Custom podAntiAffinity can be difficult to get right

**Applicable issues**

  - fixes #2636
  - fixes #2637 

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

### CODE REVIEW

File: bitnami/prometheus-operator/templates/prometheus/serviceaccount.yaml [hunk 1]
```diff
@@ -4,5 +4,8 @@ kind: ServiceAccount
 metadata:
   name: {{ template "prometheus-operator.prometheus.serviceAccountName" . }}
   labels: {{- include "prometheus-operator.prometheus.labels" . | nindent 4 }}
+  {{- with .Values.prometheus.serviceAccount.annotations }}
+  annotations: {{- toYaml . | nindent 4 }}
+  {{- end }}
```
Comment (juan131) [inline bitnami/prometheus-operator/templates/prometheus/serviceaccount.yaml:9]: Please use this approach instead that accepts templates in the annotations:
```suggestion
  {{- if .Values.prometheus.serviceAccount.annotations }}
  {{- include "prometheus-operator.tplValue" (dict "value" .Values.prometheus.serviceAccount.annotations "context" $) | nindent 4 }}
  {{- end }}
```
```

File: bitnami/prometheus-operator/templates/prometheus/service.yaml [hunk 2]
```diff
@@ -9,6 +9,9 @@ metadata:
   {{- end }}
 spec:
   type: {{ .Values.prometheus.service.type }}
+  {{- if index .Values.prometheus "stickySessions" }}
+  sessionAffinity: ClientIP
+  {{- end }}
```
Comment (juan131) [inline bitnami/prometheus-operator/templates/prometheus/service.yaml:14]: You need to add this `prometheus.stickySessions` parameter to the **values.yaml** and the **values-production.yaml**

File: bitnami/prometheus-operator/templates/prometheus/prometheus.yaml [hunk 3]
```diff
@@ -40,7 +40,9 @@ spec:
   {{- else if .Values.prometheus.replicaExternalLabelName }}
   replicaExternalLabelName: "{{ .Values.prometheus.replicaExternalLabelName }}"
   {{- end }}
-  {{- if and .Values.prometheus.ingress.enabled .Values.prometheus.ingress.hosts }}
+  {{- if .Values.prometheus.externalUrl }}
+  externalUrl: "{{ .Values.prometheus.externalUrl }}"
+  {{- else if and .Values.prometheus.ingress.enabled .Values.prometheus.ingress.hosts }}
```
Comment (juan131) [inline bitnami/prometheus-operator/templates/prometheus/prometheus.yaml:45]: You need to add this `prometheus. externalUrl ` parameter to the **values.yaml** and the **values-production.yaml**

**Comment by Fryuni** [comment]
Since you are changing this, could you add a parameter to `routePrefix` with the default to `"/"`?  
Currently it is hardcoded, I was just thinking of opening a PR to change the `externalUrl` and the `routePrefix`

**Comment by Nuru** [comment]
@miguelaeh wrote:
> We need to wait until the CI/CD updates the tags of the images.

I do not understand. What updates are needed? I did not change any tags or references to images, only the chart version, per Bitnami's request.

**Comment by miguelaeh** [comment]
Our CI/CD updates always the tags on the images before merging the PRs. For example see this https://github.com/bitnami/charts/pull/2654#issuecomment-633864242

Even if you don't modify them, the CI/CD will replace the tags by the latest ones.

**Comment by miguelaeh** [comment]
Hi @Nuru ,
I am sorry for the time this is taking, it seems we have not permissions to edit your fork.
In order to update the image tags to point to latest releases, our CI/CD needs to modify you PR (bitnami-bot will update values.yaml and values-production.yaml for you). To do so, it's necessary you allow "edits from maintainers". Find more information in the link below:

    https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by Nuru** [comment]
> Hi @Nuru ,
> I am sorry for the time this is taking, it seems we have not permissions to edit your fork.
> In order to update the image tags to point to latest releases, our CI/CD needs to modify you PR (bitnami-bot will update values.yaml and values-production.yaml for you). To do so, it's necessary you allow "edits from maintainers". 

Sorry, @miguelaeh but that option is not available for organization-owned repos/forks such as this one. If you give me the GitHub ID of your CI/CD bot I can add it as a collaborator (I think) or I can just make the change for you.

**Comment by miguelaeh** [comment]
Hi @Nuru ,
It would be great if you could add `bitnami-bot` to the collaborators.

**Comment by Nuru** [comment]
@miguelaeh I invited bitnami-bot to collaborate. 

FYI, the way we handle this for our public repos is that our bots generate pull requests against the PR branch, so the PR owner can accept that PR and merge the bot's changes into the PR.

**Pull request merged**